lib/spack/spack/spec.py

```diff
@@ -753,35 +754,83 @@ def __str__(self):
         return "{deps: %s}" % ', '.join(str(d) for d in sorted(self.values()))
 
 
-def _libs_default_handler(descriptor, spec, cls):
-    """Default handler when looking for 'libs' attribute. The default
-    tries to search for 'lib{spec.name}' recursively starting from
-    `spec.prefix`.
+def _executable_default_handler(descriptor, spec, cls):
+    """Default handler when looks for 'executable' attribute. The default
+    tries to search for {spec.name} in the ``spec.prefix.bin`` directory.
 
     :param ForwardQueryToPackage descriptor: descriptor that triggered
         the call
     :param Spec spec: spec that is being queried
     :param type(spec) cls: type of spec, to match the signature of the
         descriptor `__get__` method
     """
-    name = 'lib' + spec.name
-    shared = '+shared' in spec
```

Comment:
@alalazo Previously, any package without a `shared` variant would search for static libraries. I changed this behavior below.